### ISSUE TITLE
Authentication required for scraping prometheus metrics in emqx v4.4.10

### ISSUE THREAD

**Issue by zoeponta** [emqx/emqx#9287]
### What happened?

According to #8955, I thought authentication was no longer required, but it was.

Am I mistaken?

### What did you expect to happen?

Authentication for `api/v4/emqx_prometheus` was not required.

### How can we reproduce it (as minimally and precisely as possible)?

<details>

call `api/v4/emqx_prometheus` without authentication

```console
$ curl -i http://127.0.0.1:8081/api/v4/emqx_prometheus?type=prometheus
HTTP/1.1 401 Unauthorized
WWW-Authenticate: Basic Realm="minirest-server".
content-length: 12
Date: Wed, 02 Nov 2022 12:34:49 GMT
server. Cowboy

UNAUTHORIZED.
```

call `api/v4/emqx_prometheus` with authentication

```console
$ curl -i http://admin:public@127.0.0.1:8081/api/v4/emqx_prometheus?type=prometheus
http/1.1 200 ok
Content Length: 10685
Content Type: text/plain
Date: Wed, 02 Nov 2022 12:34:54 GMT
Connect to the server. Cowboy.

# TYPE erlang_vm_ets_limit gauge
erlang_vm_ets_limit 2097152
# TYPE erlang_vm_logical_processors gauge
erlang_vm_logical_processors 4
...
```

</details>

### EMQX version

<details>

```console
/opt/emqx $ emqx_ctl broker
sysdescr  : EMQ X Broker
version   : 4.4.10
uptime    : 42 minutes, 19 seconds
datetime  : 2022-11-02 12:43:13
```

</details>

### OS version

Docker image: emqx:4.4.10

**Comment by zmstone** [comment]
Hi @zoeponta 

Thank you for reporting the issue.

The same endpoint is available from the 18083 (dashboard) port.
We'll make the same fix for 8081 port too.
```
$ curl -s "http://127.0.0.1:18083/api/v4/emqx_prometheus?type=prometheus"  | head
# TYPE erlang_vm_ets_limit gauge
erlang_vm_ets_limit 256000
# TYPE erlang_vm_logical_processors gauge
erlang_vm_logical_processors 20
# TYPE erlang_vm_logical_processors_available gauge
```

**Comment by zoeponta** [comment]
Thank you for your reply.

I could call api without authentication on dashboard endpoint.

> We'll make the same fix for 8081 port too.

I don't use the dashboard plugin, so it's nice to call it without authentication on management endpoint.

**Comment by tigercl** [comment]
Hi @zoeponta 

Sorry to keep you waiting, we have fixed this issue in [v4.4.11](https://github.com/emqx/emqx/releases/tag/v4.4.11).

**Comment by JimMoen** [comment]
Fixup released.
